projects/knora-ui/src/lib/viewer/values/decimal-value/decimal-value.component.ts

```diff
@@ -0,0 +1,102 @@
+import {Component, Inject, Input, OnChanges, OnDestroy, OnInit, SimpleChanges} from '@angular/core';
+import {BaseValueComponent} from '../base-value.component';
+import {CreateDecimalValue, ReadDecimalValue, UpdateDecimalValue} from '@knora/api';
+import {Subscription} from 'rxjs';
+import {FormBuilder, FormControl, FormGroup, Validators} from '@angular/forms';
+
+@Component({
+  selector: 'kui-decimal-value',
+  templateUrl: './decimal-value.component.html',
+  styleUrls: ['./decimal-value.component.scss']
+})
+export class DecimalValueComponent extends BaseValueComponent implements OnInit, OnChanges, OnDestroy {
+  
+  @Input() displayValue?: ReadDecimalValue;
+
+  valueFormControl: FormControl;
+  commentFormControl: FormControl;
+
+  form: FormGroup;
+
+  valueChangesSubscription: Subscription;
+
+  customValidators = [Validators.pattern(/^-?[0-9]+[.]{1}[0-9]+$/)]; // only allow decimal values
```

Comment:
You actually do not need a custom validator here: also an integer is valid decimal (but not every decimal is a valid integer). Since the template only allows for numeric values, we should be safe.

It might be worth to look at the allowed 'e' input: https://stackoverflow.com/questions/31706611/why-does-the-html-input-with-type-number-allow-the-letter-e-to-be-entered-in

Reply:
I think allowing the user to type "e" is okay. The form control seems to convert it into the actual number before it is sent to Knora accepts the value. The only thing is that maybe the user won't like that it gets converted automatically meaning they will never be able to type "2e2" and get "2e2" back. It will always return the convert value, in this case 200, because it is converted before it is sent to Knora.